### PR TITLE
Use a secret instead of a configmap for alpha config

### PR DESCRIPTION
**Pull request by mkjpryor**
This PR addresses the concerns in #104, namely that using the alpha config requires storing the client secret in a configmap, by moving the whole alpha config to a secret.

Once we have a better way of locating secrets, this can be changed.

### CODE REVIEW

**Comment by mkjpryor** [comment]
Note that the patch version was bumped rather than the minor version as:

  1. The change is backwards compatible
  2. I consider exposing secret information to be a bug

**Comment by pierluigilenoci** [comment]
@mkjpryor, could you please rebase?

**Comment by mkjpryor** [comment]
@pierluigilenoci rebase done

**Review by pierluigilenoci — APPROVED** [review]

**Comment by pierluigilenoci** [comment]
@jessebot I was waiting for a review from @JoelSpeed.
Ref: https://github.com/oauth2-proxy/manifests/issues/104#issuecomment-1233395397

**Pull request merged**
